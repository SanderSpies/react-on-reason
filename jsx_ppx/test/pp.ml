open Migrate_parsetree

let () =
  Driver.register
    ~name:"JSX"
    Migrate_parsetree.Versions.ocaml_404 (fun _config _cookies -> Reactjs_jsx_ppx_v3_lib.jsxMapper ())
(* To run as a standalone binary, run the registered drivers *)
let () = Driver.run_main ()