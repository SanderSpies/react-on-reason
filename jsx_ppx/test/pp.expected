let div ~children  () = children
module Test =
  struct
    external makeProps :
      a:'a -> b:'b -> ?key:'key -> unit -> < a :'a ;b :'b  > Js.t = ""
    [@@bs.obj ]
    let make =
      let Test (Props : < a :'a ;b :'b  > Js.t) =
        let a = ## Props a in
        let b = ## Props b in
        print_endline (a ^ b);
        print_endline "This function should be named `Test`";
        ReactDOMRe.createElement "div" [||] in
      Test
  end
